src/content/1.7/modules/sample-modules/extending-sf-form-with-upload-image-field.md

```diff
@@ -0,0 +1,586 @@
+---
+title: Extending Symfony form with upload image field
+weight: 3
+aliases: 
+    - /1.7/modules/sample_modules/grid-and-identifiable-object-form-hooks-usage
+---
+
+# Extending Symfony form with upload image field
+{{< minver v="1.7.7" title="true" >}}
+
+
+## Introduction
+
+In this tutorial we are going to build a module which extends `Suppliers` form 
+(SELL -> Catalog -> Brands & Suppliers). You will learn how to create:
+
+- Main module class
+- Installer class
+- Create Symfony controller
+- Doctrine entity
+- Repository class
+- Image Uploader class
+- Twig View template
+
+### Main module class
+
+Let's create main module class `DemoExtendSymfonyForm2`
+
+```php
+<?php
+declare(strict_types=1);
```

Comment:
```suggestion
// since this module is compatible with PS 1.7.7 and later, we can use PHP7 strict types because PHP5 support has been dropped for PS 1.7.7
declare(strict_types=1);
```